server/scheduling/scheduler.ts

```diff
@@ -20,21 +20,29 @@ function dateDiff(dateOne: Date, dateTwo: Date) : number {
 	return dateTwo.getTime() - dateOne.getTime();
 }
 
-export async function scheduleJob(postData: Partial<IPost>, fn:(postManager: PostManager, argsObj: Partial<IPost>) => void, id?: mongoose.Types.ObjectId): Promise<void> {
+export async function scheduleJob(postData: Partial<IPost>, fn: (postManager: PostManager, argsObj: Partial<IPost>) => Promise<void>, id?: mongoose.Types.ObjectId): Promise<boolean> {
 	let time = dateDiff(new Date(), postData.date);
 	time = Math.max(time, 0);
 	try {
 		if (!id) {
 			id = await jobManager.addJob(postData, fn);
 		}
 
-		lt.setTimeout(() => {
+		lt.setTimeout(async() => {
 			// as fn is called with eval we have to add the data to its local scope
-			fn(postManager, postData);
-			jobManager.deleteJob(id);
+			try {
+				await fn(postManager, postData);
+				jobManager.deleteJob(id);
```

Comment:
I'm just curious, if the post has been added successfully, but deleteJob fails. So this job will be run again later and will save one post post?
Or if adding the post fails?